Sorbet/README.md

```diff
@@ -0,0 +1,113 @@
+
+> Q. Are there other forms of type test? If so, explain.
+
+* `T.assert_type!(x, T)`: Asserts `x` has type `T`, raising an error if not, and refining the type in the current scope.
+* `case` statements with type checks: Sorbet can narrow types based on `when` clauses using `is_a?`.
+
+<https://sorbet.org/docs/type-assertions>
+
+> Q. How do type casts work in this language?
+
+`T.let(x, T)` and `T.cast(x, T)` assign type `T` to `x`. `T.let` is used for local type annotations, while `T.cast` inserts runtime checks (in strict mode) to enforce the type. Both update the type environment statically.
+
+Example: `T.cast(x, String)` ensures `x` is treated as `String`.
+
+<https://sorbet.org/docs/type-assertions>
+
+> Q. What is the syntax for a symmetric (2-way) type-narrowing predicate?
+
+Sorbet does not support type-narrowing predicates (symmetric or otherwise). Functions returning `T::Boolean` (e.g., `x.is_a?(String)`) do not refine the argument’s type in the caller’s scope.
+
+Example: A function like `sig { params(x: T.any(String, Integer)).returns(T::Boolean) }` cannot refine `x` to `String` or `Integer` based on the return value.
+
+<https://sorbet.org/docs/from-typescript>
+
+> Q. If the language supports other type-narrowing predicates, describe them below.
+
+Sorbet lacks support for type-narrowing predicates, including one-way or checked predicates. The benchmark implements these as stubs with dummy returns (`true`, `0`) for success cases and type errors (e.g., `x.is_nan`) for failure cases to highlight this limitation.
+
+## Benchmark Details
+
+> Q. Are any benchmarks inexpressible? Why?
+
+No, all benchmarks are expressible in Sorbet. However, predicate-related benchmarks (`predicate_2way`, `predicate_1way`, `predicate_checked`) are implemented as stubs due to Sorbet’s lack of type predicate support.
+
+> Q. Are any benchmarks expressed particularly well, or particularly poorly? Explain.
+
+* **Well-expressed**: `positive`, `negative`, `alias`, `connectives`, `nesting_body`, `nesting_condition` use straightforward `is_a?` checks and align closely with Ruby’s dynamic typing idioms, making them natural for Sorbet’s flow-sensitive typing.
+* **Poorly-expressed**: `predicate_2way`, `predicate_1way`, `predicate_checked` are stubs because Sorbet cannot express type predicates. Success cases return dummy values, and failure cases trigger artificial type errors (e.g., `x.is_nan`).
+
+> Q. How direct (or complex) is the implementation compared to the pseudocode from If-T?
+
+The implementation is direct for most benchmarks, using Ruby’s `if`/`else` and `is_a?` to mirror If-T’s pseudocode. Exceptions:
+* Predicate benchmarks use stubs instead of predicate logic.
+* `object_properties` uses Hashes instead of objects, as Ruby’s structs are less common.
+* `tuple_length` uses union types (`T.any([Integer, Integer], [String, String, String])`) to encode size distinctions, which is less direct but necessary for static typing.
+
+## Advanced Examples
```

Comment:
Oh no, this README section is talking about code from `EXAMPLES.md` , which doesnt exist for Ruby yet. It's not about harder programs from the main benchmark.

We'll open an issue for If-T about the unclear instructions.